test/integration/bastion/bastion_test.go

```diff
@@ -0,0 +1,578 @@
+		WorkerSecurityGroupID: workerSecurityGroupID,
+	}
+}
+
+func teardownInfrastructure(ctx context.Context, logger *logrus.Entry, awsClient *awsclient.Client, infra *infrastructure) {
+	Expect(integration.DestroySubnet(ctx, logger, awsClient, infra.SubnetID)).To(Succeed())
+	Expect(integration.DestroySecurityGroup(ctx, logger, awsClient, infra.WorkerSecurityGroupID)).To(Succeed())
+	Expect(integration.DestroyVPC(ctx, logger, awsClient, infra.VPCID)).To(Succeed())
+}
+
+func setupShootEnvironment(ctx context.Context, logger *logrus.Entry, c client.Client, namespace *corev1.Namespace, cluster *extensionsv1alpha1.Cluster) {
+	Expect(c.Create(ctx, namespace)).To(Succeed())
+	Expect(c.Create(ctx, cluster)).To(Succeed())
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v1beta1constants.SecretNameCloudProvider,
+			Namespace: namespace.Name,
+		},
+		Data: map[string][]byte{
+			aws.AccessKeyID:     []byte(*accessKeyID),
+			aws.SecretAccessKey: []byte(*secretAccessKey),
+		},
+	}
+	Expect(c.Create(ctx, secret)).To(Succeed())
+}
+
+func teardownShootEnvironment(ctx context.Context, logger *logrus.Entry, c client.Client, namespace *corev1.Namespace, cluster *extensionsv1alpha1.Cluster) {
+	Expect(client.IgnoreNotFound(c.Delete(ctx, namespace))).To(Succeed())
+	Expect(client.IgnoreNotFound(c.Delete(ctx, cluster))).To(Succeed())
+}
+
+func setupBastion(ctx context.Context, logger *logrus.Entry, awsClient *awsclient.Client, c client.Client, name string, cluster *controller.Cluster) (*extensionsv1alpha1.Bastion, *bastionctrl.Options) {
+	bastion, err := newBastion(name)
+	Expect(err).NotTo(HaveOccurred())
+
+	options, err := bastionctrl.DetermineOptions(ctx, bastion, cluster, awsClient)
+	Expect(err).NotTo(HaveOccurred())
+
+	Expect(c.Create(ctx, bastion)).To(Succeed())
+
+	return bastion, options
+}
+
+func teardownBastion(ctx context.Context, logger *logrus.Entry, c client.Client, bastion *extensionsv1alpha1.Bastion) {
+	By("delete bastion")
+	Expect(client.IgnoreNotFound(c.Delete(ctx, bastion))).To(Succeed())
+
+	By("wait until bastion is deleted")
+	err := extensions.WaitUntilExtensionCRDeleted(
+		ctx,
+		c,
+		logger,
+		func() extensionsv1alpha1.Object { return &extensionsv1alpha1.Bastion{} },
+		extensionsv1alpha1.BastionResource,
+		bastion.Namespace,
+		bastion.Name,
+		10*time.Second,
+		16*time.Minute,
+	)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func newCluster(name string) (*extensionsv1alpha1.Cluster, *controller.Cluster) {
+	providerConfig := &awsv1alpha1.CloudProfileConfig{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "aws.provider.extensions.gardener.cloud/v1alpha1",
+			Kind:       "CloudProfileConfig",
+		},
+		MachineImages: []awsv1alpha1.MachineImages{
+			{
+				Name: "ubuntu",
+				Versions: []awsv1alpha1.MachineImageVersion{
+					{
+						Version: "20.04.20210223",
+						Regions: []awsv1alpha1.RegionAMIMapping{
+							{
+								Name: *region,
+								AMI:  "ami-08bac620dc84221eb",
+							},
```

Comment:
@rfranzke how is this done for other tests that need to spin up a machine? are the values (version, ami) hardcoded or dynamically fetched

Reply:
I presume we have to fetch it, as tests are probably not running in eu-west-1. I was unsure at first, so I made the simplest solution for now. ^^

Reply:
@rfranzke can you clarify your thumbs up emoji? ^^

Reply:
You already wrote it:

> I presume we have to fetch it, as tests are probably not running in eu-west-1.

As the region is configurable (and probably should be) the AMI needs to be fetched either dynamically or also configured by the test executor.

Reply:
> how is this done for other tests that need to spin up a machine? are the values (version, ami) hardcoded or dynamically fetched

all other tests currently have a configuration where the test executor passes the needed values.
For the shoot integration tests we use the `testrunner` that fetches the latest k8s version from the cloudprofile and passes it to the test. (samme for the os version).

Reply:
The AMI is now determined dynamically based on the AMI name. From my tests, it seems the `ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20210223` AMI is available in all regions.

Reply:
as the hardcoded image is only for the test and the ami is dynamically fetched based on the region this is fine for the test.
cc @dguendisch